### PR TITLE
CSS Selectors Steps

### PR DESCRIPTION
**Pull request by simonsipl**
- first iteration for implementing css selectors inside steps without declaring it in page object
- implementing checker for page object/css selector in step
- tests for new mechanism

### CODE REVIEW

**Comment by simonsipl** [comment]
There will be one more PR with handlers, as this would be unclear to review all of this in one PR @tgorskievo

File: src/pages/base.ts [hunk 1]
```diff
@@ -49,43 +50,63 @@ class Page {
...
+    return waitForVisibilityOf(this.getElement(elementName));
   }
 
   public waitForInvisibilityOf(elementName) {
-    return waitForInvisibilityOf(this[elementName]);
+    return waitForInvisibilityOf(this.getElement(elementName));
+  }
+
+  public getElement(elementName) {
+    if (!this[elementName]) {
```
Comment (tgorskievo) [inline src/pages/base.ts]: I think that will be more readable if you change  - `if true` then do something instead of `if false`.

Comment (simonsipl) [reply src/pages/base.ts]: It's done on purpose :)

**Comment by simonsipl** [comment]
Documentation will be done in separate PR for this changes @sethii @tgorskievo as it will be too big and there might occur problems with clear review.

File: functional-tests/features/forms/fill_and_check_form_css.feature [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    As a kakunin user
+    I want fill and check form fields
+
+    Scenario: Fill and check form fields
+        Given I visit the "main" page
+        When I click the "formLink" element
+        Then the "simpleForm" page is displayed
+        When I generate random "stringWithLength:10" as "storedStringWithLength"
+        And I fill the "form" form with:
+            | nameInput | v:storedStringWithLength |
```
Comment (tgorskievo) [inline functional-tests/features/forms/fill_and_check_form_css.feature]: You forget to set `nameInput`to css selector

File: functional-tests/features/matchers/match_current_date_css.feature [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Feature: Matchers
+    As a kakunin user
+    I want to navigate to matcher page and match current date
+
+    Scenario: I want to match current date with format
+        Given I visit the "main" page
+        When I click the "matchersLink" element
```
Comment (tgorskievo) [inline functional-tests/features/matchers/match_current_date_css.feature]: You forget to set `matchersLink` to css selector

File: functional-tests/features/matchers/match_current_date_css.feature [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        And there is element "span.current_date" with value "f:isVisible"
+        And there is element "p.date-matcher" with value "t:Date/Time:"
+        And there is element "span.current_date" with value "f:isClickable"
+        And there is element "span.current_date" with value "f:isPresent"
+        And there is element "span.current_date" with value "r:notEmpty"
+        And there is element "span.current_date" with value "f:currentDate:YYYY-MM-DD"
+
+    Scenario: I want to match current date without additional parameters
+        Given I visit the "main" page
+        When I click the "matchersLink" element
```
Comment (tgorskievo) [inline functional-tests/features/matchers/match_current_date_css.feature]: You forget to set `matchersLink` to css selector

File: functional-tests/features/matchers/matchers_css.feature [hunk 5]
```diff
@@ -0,0 +1,21 @@
+Feature: Matchers
+    As a Kakunin user
+    I want to fill input and then check if the value matches the expected result
+
+    Scenario: Fill the input and check value
+        Given I visit the "main" page
+        When I click the "formLink" element
```
Comment (tgorskievo) [inline functional-tests/features/matchers/matchers_css.feature]: You forget to set `formLink` to css selector

File: functional-tests/features/matchers/matchers_css.feature [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        Then the "simpleForm" page is displayed
+        And I fill the "form" form with:
```
Comment (tgorskievo) [inline functional-tests/features/matchers/matchers_css.feature]: You forget to set `form` to css selector

File: functional-tests/features/matchers/matchers_css.feature [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+            | nameInput | test |
```
Comment (tgorskievo) [inline functional-tests/features/matchers/matchers_css.feature]: You forget to set `nameInput` to css selector

File: functional-tests/features/wait-for-elements/wait_for_form_css.feature [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        Then the "appearSimpleForm" page is displayed
+        When I click the ".colored" element
+        And I fill the "form" form with:
+            | input[name="name"]           | d:test-dictionary:test-name |
+            | textarea[name="description"] | some description            |
+            | input[type="checkbox"]       | Checkbox Option 2           |
+            | input[type="checkbox"]       | Checkbox Option 3           |
+            | input[type="radio"]          | third-radio-option          |
+            | select[name="status"]        | unknown                     |
+        Then the "form" form is filled with:
```
Comment (tgorskievo) [inline functional-tests/features/wait-for-elements/wait_for_form_css.feature]: You forget to set `form` to css selector

File: functional-tests/features/wait-for-elements/wait_for_form_css.feature [hunk 9]
```diff
@@ -0,0 +1,25 @@
+Feature: Wait for forms
+    As a kakunin user
+    I want fill and check form fields
+
+    Scenario: Fill and check form fields
+        Given I visit the "main" page
+        When I click the ".appearForm" element
+        Then the "appearSimpleForm" page is displayed
+        When I click the ".colored" element
+        And I fill the "form" form with:
```
Comment (tgorskievo) [inline functional-tests/features/wait-for-elements/wait_for_form_css.feature]: You forget to set `form` to css selector

Comment (simonsipl) [reply functional-tests/features/wait-for-elements/wait_for_form_css.feature]: it's the name of element ;) and css selector (related to all forms)

**Pull request merged**